### PR TITLE
Add a new workspace creation URL parameter

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Add a new workspace creation URL parameter named ```editor-image```

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/22733

### Is it tested? How?

1. Deploy Eclipse-Che with the image from this PR:
<details>
<summary>kubectl patch command</summary>

```bash
kubectl patch -n eclipse-che "checluster/eclipse-che" --type=json -p="[{"op": "replace", "path": "/spec/components/dashboard/deployment", "value": {containers: [{image: "quay.io/eclipse/che-dashboard:pr-1037", name: che-dashboard}]}}]"
```
</details>

2. Create a new workspace with the custom editor image from the next link
```bash
{CHE-server}#https://github.com/eclipse-che/che-dashboard&editor-image=quay.io/mloriedo/che-code:copilot-builtin
```

3. The new workspace should be created with the custom editor image ```quay.io/mloriedo/che-code:copilot-builtin```.
It  could be checked with **swagger**
```bash
 {CHE-server}/dashboard/api/swagger
 ```
Знімок екрана 2024-01-16 о 16 52 26

4. Create a new workspace with the custom che-editor and editor image from the next link
```bash
{CHE-server}#https://github.com/che-samples/bash&che-editor=che-incubator/che-idea/next&editor-image=quay.io/mloriedo/che-rider:2023:2
```
5. The new workspace should be created with the custom editor ```che-incubator/che-idea/next``` and the custom editor image ```quay.io/mloriedo/che-rider:2023:2```.
It  could be checked with **swagger**
```bash
 {CHE-server}/dashboard/api/swagger
 ```

Знімок екрана 2024-01-16 о 17 24 50

### CODE REVIEW

File: packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/editorImage.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    ];
+  if (!devfileSourceYaml) {
+    return undefined;
+  }
+  const factorySearchParams = (
+    load(devfileSourceYaml) as {
+      factory?: {
+        params?: string;
+      };
+    }
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/editorImage.ts]: I have fixed it.

File: packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/editorImage.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  } catch (err) {
+    throw new Error(`Failed to update editor image. ${common.helpers.errors.getMessage(err)}`);
+  }
+}
+
+export function getEditorImage(workspace: devfileApi.DevWorkspace): string | undefined {
+  const devfileSourceYaml =
+    workspace.spec.template.attributes?.['dw.metadata.annotations']?.[
+      'che.eclipse.org/devfile-source'
+    ];
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/editorImage.ts]: I have fixed it.

File: packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/index.ts [hunk 3]
```diff
@@ -533,6 +539,7 @@ export const actionCreators: ActionCreators = {
     (
       devWorkspaceResource: devfileApi.DevWorkspace,
       devWorkspaceTemplateResource: devfileApi.DevWorkspaceTemplate,
+      attributes: Partial<FactoryParams>,
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/index.ts]: I have fixed it.

**Comment by artaleks9** [comment]
Verified on Eclipse Che with `quay.io/eclipse/che-dashboard:pr-1037` - the functionality works as expected.
.

**Review by artaleks9 — APPROVED** [review]

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**